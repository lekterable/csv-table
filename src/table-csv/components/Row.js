import React from 'react'
import Cell from './Cell'

export default ({ header, cells }) => {
  return (
    <tr>
      {cells.map((cell, index) => (
        <Cell key={index} value={cell} header={header} />
      ))}
    </tr>
  )
}
