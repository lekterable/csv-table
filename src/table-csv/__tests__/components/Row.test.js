import React from 'react'
import renderer from 'react-test-renderer'
import { cleanup, render } from 'react-testing-library'
import Row from '../../components/Row'

describe('<Row />', () => {
  const withTable = component => (
    <table>
      <tbody>{component}</tbody>
    </table>
  )

  afterEach(cleanup)

  it('should match snapshot', () => {
    const component = renderer.create(<Row />).toJSON()

    expect(component).toMatchSnapshot()
  })

  it('should render row', () => {
    const { queryByTestId } = render(withTable(<Row />))
    const headerRow = queryByTestId('table-header-row')
    const row = queryByTestId('table-row')

    expect(headerRow).toBeNull()
    expect(row).not.toBeNull()
  })

  it('should render header row if `header` prop', () => {
    const { queryByTestId } = render(withTable(<Row header />))
    const headerRow = queryByTestId('table-header-row')
    const row = queryByTestId('table-row')

    expect(headerRow).not.toBeNull()
    expect(row).toBeNull()
  })

  it('should not render <Cell /> without data', () => {
    const { queryByTestId } = render(withTable(<Row />))
    const cell = queryByTestId('table-cell')

    expect(cell).toBeNull()
  })

  it('should render <Cell /> with data', () => {
    const { queryByTestId } = render(
      withTable(<Row cells={['Value 1', 'Value 2']} />)
    )
    const cell = queryByTestId('table-cell')

    expect(cell).not.toBeNull()
  })
})
