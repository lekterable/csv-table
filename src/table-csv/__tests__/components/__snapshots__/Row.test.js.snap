// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Row /> should match snapshot 1`] = `
<tr
  className="table__row"
  data-testid="table-row"
/>
`;
